// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

syntax = "proto3";

package spoor.instrumentation;

import "google/protobuf/timestamp.proto";

message FunctionInfo {
  string linkage_name = 1;
  string demangled_name = 2;
  string file_name = 3;
  string directory = 4;
  int32 line = 5;
  bool instrumented = 6;
};

message InstrumentedFunctionMap {
  string module_id = 1;
  map<uint64, FunctionInfo> function_map = 2;
  google.protobuf.Timestamp created_at = 3;
}
